### DISCUSSION TITLE
Why use operator<< instead of fmt::formatter for custom types?

### DISCUSSION THREAD

**Discussion by Codeazure** [odygrd/quill#188] [Q&A] [answered]
Why did you decide to use ostream operator<< for converting custom types into strings and not fmt::formatter? The formatter is much faster & more consistent since it means we can use fmt formatting everywhere. Is there a technical reason why use of fmt::formatter is not a good idea?

**Comment by odygrd** [comment, marked as answer]
There isn't a particular reason for using ostream operator<< in the examples. I just did it for convenience as I assumed that most people know about ostream overloading compared to the fmt formatter one. 
fmt formatter should also work with the examples so it is up to the user what to use.

This has now been added to the documentation https://quillcpp.readthedocs.io/en/latest/tutorial.html#requirements
